src/h5web/vis-packs/core/hooks.ts

```diff
@@ -50,16 +45,15 @@ export function useDatasetValue(
   return valuesStore.get({ path: dataset.path, selection });
 }
 
-export function useDatasetValues<S extends HDF5Shape, T extends HDF5Type>(
-  datasets: Dataset<S, T>[]
-) {
+export function useDatasetValues<D extends Dataset>(
+  datasets: D[]
+): Record<string, Value<D>>;
+
+export function useDatasetValues(datasets: Dataset[]): Record<string, unknown> {
   const { valuesStore } = useContext(ProviderContext);
 
   return Object.fromEntries(
-    datasets.map(({ name, path }) => [
-      name,
-      valuesStore.get({ path }) as Value<S, T>,
-    ])
+    datasets.map(({ name, path }) => [name, valuesStore.get({ path })])
```

Comment:
Here I'm replacing the type cast with a return type of `Record<string, unknown>` and an override with the return type that the caller cares about, `Record<string, Value<D>>`, thus clearly separating the typing of the function's internal declaration and external usage.